### ISSUE TITLE
Sample Android app code crashes when I touch the touchscreen

### ISSUE THREAD

**Issue by Digital-Larry** [grame-cncm/faust#321]
From: https://faust.grame.fr/doc/tutorials/#turn-it-into-an-android-app

--
import("stdfaust.lib");
f = hslider("freq[acc: 0 0 -10 0 10]",1000,50,2000,0.01) : si.smoo;
phasor(freq) = (+(freq/ma.SR) ~ ma.decimal);
osc(freq) = sin(phasor(freq)*2*ma.PI);
organ(freq) = (osc(freq) + osc(freq*2) + osc(freq*3))/3;
process = organ(f)/3;

--
Export as android/smartkeyb

https://faustservicecloud.grame.fr/CE024C7A4E36A8D65941392DDC48FFA25D3DDBCF/android/smartkeyb/binary.apk

The program opens and displays a black and white keyboard and makes a warbling tone and the pitch is following one axis of rotation.  However if I tap the screen the application crashes it says "unfortunately, untitled has stopped".  It behaves the same way on Android 7.0 and 4.4.2.

**Comment by sletz** [comment]
The given code does not compile: for instance freq2 and freq3 are not defined. Can you paste a working version?

**Comment by Digital-Larry** [comment]
Sorry, the copy/paste left out some characters!

import("stdfaust.lib");
f = hslider("freq",440,50,2000,0.01);
g = hslider("gain",1,0,1,0.01);
t = si.smoo(button("gate"));
phasor(freq) = (+(freq/ma.SR) ~ ma.decimal);
osc(freq) = sin(phasor(freq)*2*ma.PI);
organ(freq) = (osc(freq) + osc(freq*2) + osc(freq*3))/3;
process = organ(f)*g*t/3;

**Comment by sletz** [comment]
Can you try to add `declare voices "16";` in the code, and rebuild.

**Comment by Digital-Larry** [comment]
I actually got it to work initially by paying closer attention to the instructions.  If I export this as android/android, it actually works properly with the gyro/accelerometer.  The installed app has a different icon also.

android.dsp:
```
import("stdfaust.lib");
f = hslider("freq[acc: 0 0 -10 0 10]",1000,50,2000,0.01) : si.smoo;
phasor(freq) = (+(freq/ma.SR) ~ ma.decimal);
osc(freq) = sin(phasor(freq)*2*ma.PI);
organ(freq) = (osc(freq) + osc(freq*2) + osc(freq*3))/3;
process = organ(f)/3;
```
So, getting the download from the QR code after selecting android/android in the online editor worked.

If I try to run

faust2android android.dsp, then I get

```
5/17/2019
1:35 PM	Gradle sync started

1:38 PM	Gradle sync failed: CMake '3.6.0' was not found in PATH or by cmake.dir property.
				- CMake '3.10.2' found in SDK did not match requested version '3.6.0'.
				Consult IDE log for more details (Help | Show Log) (3 m 10 s 576 ms)
'''

**Comment by Digital-Larry** [comment]
I added this line to the top of the toy.dsp listing:

declare options "[nvoices:16]";

The I exported through the web editor as android/smartkeyb and downloaded the app by scanning the QR code.

Application still "unfortunately stops running".  Tested on the Android 7.0 phone.

I'd gladly try to dig deeper to debug this, but engaging "debug" in Android Studio doesn't show much of anything useful.  e.g. the application started, the application stopped.  I'm not that well versed in using Android Studio.
